public.yaml

```diff
@@ -17,6 +17,46 @@ metadata:
   name: velum-public
   labels:
     name: velum-public
+  annotations:
+    pod.beta.kubernetes.io/init-containers: '[
+      {
+        "name": "wait-for-mariadb-secrets",
+        "image": "sles12/velum:0.0",
+        "command": ["sh", "-c", "while [ ! -f /infra-secrets/mariadb-password ]; do sleep 1; done; exit 0;"],
+        "volumeMounts": [
+          {
+            "mountPath": "/infra-secrets",
+            "name": "infra-secrets"
+          }
+        ]
+      },
+      {
+        "name": "saltapi-secrets",
+        "image": "sles12/velum:0.0",
+        "command": ["sh", "-c", "if [ ! -f /infra-secrets/saltapi-password ]; then head -n 1000 /dev/random | sha256sum | cut -d\" \" -f1 > /infra-secrets/saltapi-password; fi; exit 0;"],
+        "volumeMounts": [
+          {
+            "mountPath": "/infra-secrets",
+            "name": "infra-secrets"
+          }
+        ]
+      },
+      {
+        "name": "salt-master-config",
+        "image": "sles12/velum:0.0",
+        "command": ["sh", "-c", "rmdir /salt-master-credentials/returner-credentials.conf; echo \"mysql:\" > /salt-master-credentials/returner-credentials.conf; echo -n \"  pass: \" >> /salt-master-credentials/returner-credentials.conf; cat /infra-secrets/mariadb-password >> /salt-master-credentials/returner-credentials.conf"],
+        "volumeMounts": [
```

Comment:
keep wondering if this could be simplified a bit, but to be honest it looks ok for now at least.